### PR TITLE
Flink: Backport #6660 to Flink 1.14 and 1.15

### PR DESCRIPTION
**Pull request by amogh-jahagirdar**
Backport https://github.com/apache/iceberg/pull/6660/files

### CODE REVIEW

File: flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java [hunk 1]
```diff
@@ -386,25 +386,28 @@ private void commitOperation(
...
     operation.set(FLINK_JOB_ID, newFlinkJobId);
     operation.set(OPERATOR_ID, operatorId);
+    operation.toBranch(branch);
 
     long startNano = System.nanoTime();
     operation.commit(); // abort is automatically called if this fails.
     long durationMs = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startNano);
     LOG.info(
-        "Committed {} to table: {}, checkpointId {} in {} ms",
+        "Committed {} to table: {}, branch: {}, checkpointId {} in {} ms",
```
Comment (amogh-jahagirdar) [inline flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java:407]: @stevenzwu I added this log for branch when backporting, noticed we were missing the branch after the operation actually committed. I'll add this to Flink 1.16 in a separate PR. Let me know your thoughts.

Comment (stevenzwu) [reply flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java:407]: I will be fine if you just add it in this PR. 

Can you check the diff output and make sure no diff related to this backport?
```
git diff --no-index  flink/v1.14/flink/src/ flink/v1.16/flink/src
git diff --no-index  flink/v1.15/flink/src/ flink/v1.16/flink/src
```

Comment (amogh-jahagirdar) [reply flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java:407]: Checked, verified there's no diff related to this backport.

**Comment by amogh-jahagirdar** [comment]
I rebased , saw there were some new changes and test additions. No conflicts but want to make sure tests pass after rebasing.

**Review by stevenzwu — APPROVED** [review]

**Comment by stevenzwu** [comment]
thanks @amogh-jahagirdar for the backport

**Pull request merged**